include/alpaka/queue/Traits.hpp

```diff
@@ -41,6 +41,8 @@ namespace alpaka
     //!   If the event has previously been queued, then this call will overwrite any existing state of the event.
     //!   Any subsequent calls which examine the status of event will only examine the completion of this most recent
     //!   call to enqueue.
+    //!   If a queue is waiting for an event the recently event state at the time of the API call to wait() will be
+    //!   used to release the queue.
```

Comment:
I am still unsure what this means. So the queue will be destroyed as part of the event's state? But that usually results in the callback thread joining itself, which is a bug. Does `release` mean destroy here?

Reply:
"Release Queue" means "Queue no longer waits"

Reply:
Can we use "unblock" instead? "Release" carries some connotation of ownership for me :)

Reply:
Will change it to `unblcok` because we call the queue non blocking queue.
@bernhardmgruber This is a general interface, you should at the moment you read comments here forget all your background knowledge of how something is implemented. Do not assume any callback threads or similar!